### DISCUSSION TITLE
npm integration not working

### DISCUSSION THREAD

**Discussion by uniboxx** [vanjs-org/van#253] [General]
Hi, I installed the npm package and imported it following the instructions but i get Failed to resolve module specifier "vanjs-core". importing the cdn or the downloaded file it works as expected. Anyone had the same problem with npm?

**Comment by sirenkovladd** [comment]
Are you using any bundling tools?
could you send some example?

**Reply by uniboxx** [reply]
Hi, I tryied with and without vite

**Reply by uniboxx** [reply]
I just created a simple html that import the script and the script that import the vanjs module

**Reply by sirenkovladd** [reply]
Here is a small example using vite
https://githubbox.com/yahia-berashish/vite-vanjs-ts

**Reply by uniboxx** [reply]
But i would to be able to run it without vite and with just the npm package, if possible

my script code:

//import van from 'https://cdn.jsdelivr.net/gh/vanjs-org/van/public/van-1.2.8.min.js';

import van from 'vanjs-core';

const { a, div, li, p, ul } = van.tags;

const Hello = () =>
  div(
    p('👋Hello'),
    ul(li('🗺️World'), li(a({ href: 'https://vanjs.org/' }, '🍦VanJS')))
  );

van.add(document.body, Hello());

same file with cdn works, with npm not

**Reply by sirenkovladd** [reply]
npm can't do that, it's just different runtimes

why it works with vite, because it combines the code that is executed into one js file if it is a build
or provides all imports (such as vanjs-core) with a js file when it is imported from a web page

in order to learn more about how it works
https://vitejs.dev/guide/features.html#npm-dependency-resolving-and-pre-bundling

**Reply by uniboxx** [reply]
thanks for the replyies. I just followed the get started tutorial and it seams that it is just needed to install with npm and import it as  module.

**Reply by uniboxx** [reply]
Eventually i will try it with cdn or the downloaded minified file

**Reply by uniboxx** [reply]
i tryied with vite and it works now. Thanks for gave me your time.
